### PR TITLE
Chain cause in ComputeException.translateAndThrow, add tests

### PR DESCRIPTION
**Pull request by mziccard**
This fixes #946 for the compute module

### CODE REVIEW

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/ComputeExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    expect(exceptionMock.getCause()).andReturn(cause).times(2);
+    replay(exceptionMock);
+    try {
+      ComputeException.translateAndThrow(exceptionMock);
+    } catch (BaseServiceException ex) {
+      assertEquals(ComputeException.UNKNOWN_CODE, ex.code());
+      assertEquals("message", ex.getMessage());
+      assertFalse(ex.retryable());
+      assertTrue(ex.idempotent());
+      assertEquals(cause, ex.getCause());
```
Comment (aozarov) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/ComputeExceptionTest.java]: I think you could have used `assertSame here.

**Comment by aozarov** [comment]
LGTM

**Comment by mziccard** [comment]
@aozarov I just noticed that we do very have different access modifiers for exceptions' constructors. In `dns` they are private while in other services they're public. I suggest we make the package scoped for the sake of testing. What do you think?

**Comment by aozarov** [comment]
Yes, I think consistency here is good and package scope would be a good choice.

**Pull request merged**